File: semestr-8/Jezyk C#/alexander/PCS/2023s/ID06TC1/BridgePattern/BridgePattern/BridgeShapeListToXml.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace BridgePattern
{
    public class BridgeShapeListToXml: BridgeShapeList
    {
        private TextBox field;
        private StringBuilder sb;
        public BridgeShapeListToXml(TextBox field)
        {
            this.field = field;
        }

        protected override void InitialiseProcess()
        {
            sb
            =
                new StringBuilder()
                .AppendLine("<root>")
                .AppendLine("\t<shapes>")
            ;
        }

        protected override void Proceed(ShapeBase shape)
        {
            sb.Append("\t\t").AppendLine(shape.ToXml());
        }

        protected override void FinaliseProcess()
        {
            field.Text=sb
                .AppendLine("\t</shapes>")
                .AppendLine("</root>")
                .ToString()
            ;
        }
    }
}

File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/2023s/IZ08IO3/BridgePattern/src/BridgeShapeListToXml.java

import java.util.List;
import javax.swing.JTextArea;

public class BridgeShapeListToXml extends BridgeShapeList
{
    private JTextArea field;
    private StringBuilder sb;
    public BridgeShapeListToXml(JTextArea field)
    {
        this.field=field;                
    }
    
    @Override public void initialiseProcess()
    {
        sb
        =
            new StringBuilder()
            .append("<root>").append(System.lineSeparator())
            .append("\t<shapes>").append(System.lineSeparator())
        ;
    }
    
    @Override public void proceed(ShapeBase shape)
    {
        sb.append("\t\t").append(shape.toXml()).append(System.lineSeparator());
    }
    
    @Override public void finaliseProcess()
    {
        field.setText
        (
            sb
            .append("\t</shapes>").append(System.lineSeparator())
            .append("</root>").append(System.lineSeparator())
            .toString()
        );
    }
}
